src/data_capture/video_capture/scripts/capture.py

```diff
@@ -48,15 +118,22 @@ def _is_record_callback(self, data):
         is_record = data.data
         try:
             if is_record:
-                rospy.loginfo("Starting to record video")
+                rospy.loginfo("Starting to record video")             
+
+                # Starting the periodic check on free memory, stops if exceeds 90% utilization at interval=2 seconds
+                # RepeatedTimer auto-starts, no need of rt.start()
+                rt = RepeatedTimer(2, monitorAvailableMemory, 90, self._video_recorder) 
+
                 self._video_recorder.start_recording()
+
             else:
                 rospy.loginfo("Stopped recording video")
                 self._video_recorder.stop_recording()
+                if 'rt' in locals() or 'rt' in globals():
+                    rt.stop()
```

Comment:
This is complex. I much prefer keeping things in one thread and not worrying about finding things in the namespace.

Also, why would you be unsure if it'd be in locals or globals?